pandas/tests/scalar/period/test_period.py

```diff
@@ -767,6 +768,34 @@ def test_period_deprecated_freq(self):
             assert isinstance(p1, Period)
             assert isinstance(p2, Period)
 
+    @pytest.mark.parametrize("bound, offset", [(Timestamp.min, -1), (Timestamp.max, 1)])
+    def test_start_time_and_end_time_bounds(self, bound, offset):
+        # GH #13346
+        period = Period(
+            year=bound.year,
+            month=bound.month,
+            day=bound.day,
+            hour=bound.hour,
+            minute=bound.minute,
+            second=bound.second - offset,
+            freq="us",
+        )
+        period.start_time
+        period.end_time
+        period = Period(
```

Comment:
Could you create a separate test function for this?

Reply:
done in the last PR